library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/mapper/GenericWireframeMapper.kt

```diff
@@ -17,60 +17,95 @@ import android.widget.TextView
+                checkedTextViewMapper.toGenericMapper()
+        )
+        mappers.add(
+            defaultTypeCheckCondition(Button::class.java) to
+                buttonMapper.toGenericMapper()
+        )
+        mappers.add(
+            defaultTypeCheckCondition(EditText::class.java) to
+                editTextViewMapper.toGenericMapper()
+        )
+        mappers.add(
+            defaultTypeCheckCondition(TextView::class.java) to
+                textMapper.toGenericMapper()
+        )
+        mappers.add(
+            defaultTypeCheckCondition(ImageView::class.java) to
+                imageMapper.toGenericMapper()
+        )
+        mappers.add(
+            decorViewCheckCondition() to
+                decorViewMapper.toGenericMapper()
+        )
+        mappers.add(
+            { _: View -> true } to
+                viewWireframeMapper.toGenericMapper()
+        )
+    }
+
     override fun map(view: View, systemInformation: SystemInformation):
         List<MobileSegment.Wireframe> {
-        return when {
-            SwitchCompat::class.java.isAssignableFrom(view::class.java) -> {
-                switchCompatMapper.map(view as SwitchCompat, systemInformation)
-            }
-            RadioButton::class.java.isAssignableFrom(view::class.java) -> {
-                radioButtonMapper.map(view as RadioButton, systemInformation)
-            }
-            CheckBox::class.java.isAssignableFrom(view::class.java) -> {
-                checkBoxMapper.map(view as CheckBox, systemInformation)
-            }
-            CheckedTextView::class.java.isAssignableFrom(view::class.java) -> {
-                checkedTextViewMapper.map(
-                    view as CheckedTextView,
-                    systemInformation
-                )
-            }
-            Button::class.java.isAssignableFrom(view::class.java) -> {
-                buttonMapper.map(view as Button, systemInformation)
-            }
-            EditText::class.java.isAssignableFrom(view::class.java) -> {
-                editTextViewMapper.map(view as EditText, systemInformation)
-            }
-            TextView::class.java.isAssignableFrom(view::class.java) -> {
-                textMapper.map(view as TextView, systemInformation)
-            }
-            ImageView::class.java.isAssignableFrom(view::class.java) -> {
-                imageMapper.map(view as ImageView, systemInformation)
-            }
-            else -> {
-                val viewParent = view.parent
-                if (viewParent == null ||
-                    !View::class.java.isAssignableFrom(viewParent::class.java)
-                ) {
-                    decorViewMapper.map(view, systemInformation)
-                } else {
-                    viewWireframeMapper.map(view, systemInformation)
-                }
-            }
+        return mappers.firstOrNull {
+            it.first(view)
+        }?.second?.map(view, systemInformation) ?: emptyList()
```

Comment:
I would suggest to use an explicit class instead of `Pair`, because here it is not that obvious what is `first` and `second`

Reply:
actually for keeping it simpler I would rather keep the Pairs as you cannot do a mistake when applying them. If you use `second` instead of `first` it will not compile as it will not match the signature.

Reply:
I meant to create a data class with 2 named fields. But it is minor anyway, just for reviews, because in IDE there are type suggestions, unlike here.